components/order-app/stores/fulfillment-store.ts

```diff
@@ -3,16 +3,16 @@ import { distanceFromTMS } from './order-utils';
 import DateStore from './date-store';
 
 class FulfillmentStore {
-  option: 'pickup' | 'delivery';
+  option: 'pickup' | 'delivery' = 'delivery';
   contactName: string = '';
   contactNumber: string = '';
-  numberOfGuests: number;
+  numberOfGuests: number = 0;
   specialInstructions: string = '';
 
-  deliveryLocation: google.maps.places.PlaceResult;
-  errorFromGoogle: boolean;
-  loadingMiles: boolean;
-  deliveryMiles: number;
+  deliveryLocation: google.maps.places.PlaceResult = {};
```

Comment:
This can be undefined if its a pickup order

Reply:
I didn't change the type annotation here. I just set the default value to {}. Should I set it to this instead?

```ts
  deliveryLocation: google.maps.places.PlaceResult | undefined = {};
```

Reply:
I think `deliveryLocation: google.maps.places.PlaceResult | undefined;` is ok

Reply:
This is causing further issues. 
In `components/order-app/stores/fulfillment-store.ts`, see the error in the comments
```ts
constructor() {
    makeAutoObservable(this);
    this.dateStore = new DateStore(this);

    reaction(
      () => this.deliveryLocation,
      // This line says that Argument of type 'PlaceResult | undefined' is not assignable to parameter of type 'PlaceResult'
      (googlePlacesObj) => this.handleDeliverLocationUpdate(googlePlacesObj),
    );
  }
```

So, I changed it to this

```ts
constructor() {
    makeAutoObservable(this);
    this.dateStore = new DateStore(this);

    reaction(
      () => this.deliveryLocation,
      (googlePlacesObj) => googlePlacesObj && this.handleDeliverLocationUpdate(googlePlacesObj),
    );
  }
```

Reply:
Also, in `components/order-app/checkout/stripe-utils.ts`, see the error in the comments

```ts
if (OrderStore.fulfillment.option === 'delivery') {
    // // This line says that Argument of type 'PlaceResult | undefined' is not assignable to parameter of type 'PlaceResult'.
    orderRequest.deliveryLocation = formatGooglePlacesObj(orderStore.fulfillment.deliveryLocation);
    orderRequest.deliveryFee = orderStore.registerStore.deliveryFee as number;
    orderRequest.numberOfGuests = orderStore.fulfillment.numberOfGuests;
}
```

So, I changed it to this
```ts
  if (OrderStore.fulfillment.option === 'delivery') {
    orderRequest.deliveryLocation = orderStore.fulfillment.deliveryLocation
      ? formatGooglePlacesObj(orderStore.fulfillment.deliveryLocation)
      : undefined;
    orderRequest.deliveryFee = orderStore.registerStore.deliveryFee as number;
    orderRequest.numberOfGuests = orderStore.fulfillment.numberOfGuests;
  }
```

Reply:
So, should I keep it as before or do these changes?

Reply:
Ah ok, just keep it as before then